pandas/core/internals.py

```diff
@@ -1456,6 +1456,11 @@ def where(self, other, cond, align=True, errors='raise',
             raise ValueError("where must have a condition that is ndarray "
                              "like")
 
+        # For SparseBlock, self.values is always 1D. If cond was a frame,
+        # it's 2D values would incorrectly broadcast later on.
+        if values.ndim == 1 and any(ax == 1 for ax in cond.shape):
+            cond = cond.ravel()
```

Comment:
This fixes https://github.com/pandas-dev/pandas/issues/17198, but certainly in an incorrect way.

Reply:
hmm, can you define ``.where()`` in ``SparseBlock`` and do a super call here instead?